File: algorithms/Strings/Medium/StringToInteger/VSDStringToInteger.py

def check(answer):
    if answer<-2**31:
        return -2**31
    elif answer>=2**31:
        return 2**31-1
    else:
        return answer

def myAtoi(string):
    answer=0
    sign=0
    i=0

    while(i<len(string) and string[i]==" "):
        i+=1
    if i==len(string):
        return answer 
    if string[i]=="+" and sign==0:
        sign=1
        i+=1
    if i==len(string):
        return answer
    if string[i]=="-" and sign==0:
        sign=-1 
        i+=1
    while (i<len(string)):
        if string[i].isdigit():
            answer=answer*10+int(string[i])
            i+=1
        else:
            if sign!=0:
                answer=sign*answer 
            return check(answer)
    if sign!=0:
        answer=sign*answer
    return check(answer)



if __name__=="__main__":
    s="42"
    print(myAtoi(s))

File: algorithms/Strings/Medium/StringToInteger/VSDStringToInteger.java

public class Main{

    static int atoi(String s) {
        int num = 0, i = 0, sign = 1;
        while (i < s.length() && s.charAt(i) == ' ') {
            i++;
        }
        if(i < s.length() && (s.charAt(i) == '-' || s.charAt(i) == '+')) {
            sign = s.charAt(i) == '+' ? 1 : -1;
            i++;
        }
        while (i < s.length() && Character.isDigit(s.charAt(i))) {
            int digit = s.charAt(i) - '0';
            if ((num > Integer.MAX_VALUE / 10) || ((num == Integer.MAX_VALUE / 10) && (digit > 7))) {
                return sign == 1 ? Integer.MAX_VALUE : Integer.MIN_VALUE;
            }
            num = (num * 10) + digit;
            i++;
        }
        return num * sign;
    }
    public static void main(String[] args) {
        String s="42";
        System.out.println(atoi(s));
    }
}